src/test/scala/chiselTests/QueueFlushSpec.scala

```diff
@@ -0,0 +1,181 @@
+package chiselTests
+
+import org.scalacheck._
+
+import chisel3._
+import chisel3.testers.BasicTester
+import chisel3.util._
+import chisel3.util.random.LFSR
+
+class ThingsPassThroughFlushQueueTester(elements: Seq[Int], queueDepth: Int, bitWidth: Int, tap: Int, useSyncReadMem: Boolean) extends BasicTester {
```

Comment:
can you add a scaladoc description r.e. what each test does?

Reply:
okay I'll do that